src/components/admin/ImageController.jsx

```diff
@@ -192,23 +200,31 @@ export default class ImageUploader extends BaseComponent {
     reader.readAsDataURL(fileObject.file);
   }
 
-  changeImageName(name) {
-    const newName = window.prompt('Please enter new name (and remember to keep ' +
-                                  `the extension) for ${name}:`);
-    if (!newName) {
-      return;
-    }
-    const index = this.state.files.findIndex(fileObject => fileObject.metaData.name === name);
+  changeImageNameOpen(name) {
+    this.setState({
+      changeNameOpen: true,
+      imgName: name,
+    });
+  }
+
+  changeImageName() {
+    const imgName = this.state.imgName;
```

Comment:
Are there instances in which we would want to allow the user to change the file extension? In most if not all cases you would not want to change the extension of an image file directly on the website.

Before reading this, I changed the code so that it does the following:
(1) If the user types in the correct extension, it leaves it as is;
(2) If the user does not include an extension at all, it adds the original extension to the new filename; and
(3) If the user inputs the wrong extension, it changes it to the original one.

Of course, if this is not enough, I'll add what you suggested!

Reply:
That sounds great, I think I agree the extension should never be changed on the website, I think we should do (3) a bit differently if you're just silently changing the extension though as that could be confusing, either give a notification with the snackbar or something else that the extension was changed back to the original or give them a warning that doesn't allow them to save.